### DISCUSSION TITLE
User_setup for ESP32_2432S022 2.2 inch TFT module

### DISCUSSION THREAD

**Discussion by arthurrep** [Bodmer/TFT_eSPI#3100] [Q&A - General]
I tried a number of setups from User_Setup_Select.h, but the  Ellipse drawing example gives no output on the display. Does anyone know how to make the module working?

**Comment by dart1280** [comment]
In a word... nope.

I spend the best part of a day messing around with this thing and the only way I could get it to display anything was by burning the default bin file from the archive.

Having said that... I did get the backlight to turn on and off!

So... from the supplied circuit diagram there are the following pins -
CS = IO17
RS = IO16
WR = IO4
RD = IO2
RST = TFT_RST

From here I made three assumptions -

RST should be set to -1 because TFT_RST is connected to RST (EN pin)
CS = CS so connect to IO17
As we only have 3 other pins I set MISO to -1 as we don't need data from the display

from there I tried  every combination for MOSI,SCK and DC. -

MOSI 	16	16	2	2	4	4
SCK		2	4	4	16	16	2
DC		4	2	16	4	2	16

.......... and got nothing.....

![ESP32-2432022-LCM-V1 0](https://github.com/Bodmer/TFT_eSPI/assets/79755245/d0bda10d-e6f5-499e-a027-b62b4d6ce19b)
![ESP32-2432022-MCU-V1 0](https://github.com/Bodmer/TFT_eSPI/assets/79755245/73664b00-57b8-4f59-a4b2-14b6250ab675)

Here is my simple test sketch
`
//#include <Arduino.h>
#include "TFT_eSPI.h"
//#include <esp_wifi.h>
//#include <esp_event_loop.h>
//#include <nvs_flash.h>

TFT_eSPI tft = TFT_eSPI(240, 380); // Invoke custom library


void setup(){ 

tft.init();
tft.begin();
tft.setRotation(1);


pinMode(0, OUTPUT);
digitalWrite(0, LOW);
delay(1000);
digitalWrite(0, HIGH); 
}

void loop(){
tft.fillScreen(TFT_BLACK);
    delay(1000);
    tft.fillScreen(TFT_RED);
    delay(1000);
    tft.fillScreen(TFT_GREEN);
    delay(1000);
    tft.fillScreen(TFT_BLUE);
    delay(1000);
}
`
And here is my Setup_esp32_2432s022.h
`


#define ST7789_DRIVER     // Configure all registers

#define TFT_WIDTH  240
#define TFT_HEIGHT 320

 #define TFT_BL   0            // LED back-light control pin
 #define TFT_BACKLIGHT_ON HIGH  // Level to turn ON back-light (HIGH or LOW)

//#define TFT_RGB_ORDER TFT_RGB  // Colour order Red-Green-Blue
//#define TFT_RGB_ORDER TFT_BGR  // Colour order Blue-Green-Red

//#define TFT_INVERSION_ON
//#define TFT_INVERSION_OFF

// DSTIKE stepup
//#define TFT_DC    23
//#define TFT_RST   32
//#define TFT_MOSI  26
//#define TFT_SCLK  27

// Generic ESP32 setup

#define TFT_MOSI 4
#define TFT_SCLK 2
#define TFT_DC   16


#define TFT_CS   17  //do not change
#define TFT_RST  -1  //do not change
//#define TFT_MISO 4 //do not change


// For NodeMCU - use pin numbers in the form PIN_Dx where Dx is the NodeMCU pin designation
//#define TFT_CS   17      // Define as not used
//#define TFT_DC   -1  // Data Command control pin
//#define TFT_RST  -1  // TFT reset pin (could connect to NodeMCU RST, see next line)
//#define TFT_RST  -1    // TFT reset pin connect to NodeMCU RST, must also then add 10K pull down to TFT SCK


#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
//#define LOAD_FONT8N // Font 8. Alternative to Font 8 above, slightly narrower, so 3 digits fit a 160 pixel TFT
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts

#define SMOOTH_FONT

#define SPI_FREQUENCY  27000000
//#define SPI_FREQUENCY  40000000

#define SPI_READ_FREQUENCY  20000000

#define SPI_TOUCH_FREQUENCY  2500000

// #define SUPPORT_TRANSACTIONS
`
I give up :-)

**Reply by Rust-Hero** [reply]
Yeah, the eSPI libraries incredibly difficult to work with considering the parallel interface... accommodations for the effort, but the Arduino_GFX library for the display and TAMC_GT911 for touch are significantly more compatible.....

**Comment by arthurrep** [comment]
Thank you so much for your effort. Sorry that you could not get it to work…

Is there another comparable display that does work?

 

Thanks again,

Arthur Rep

 

Van: dart1280 ***@***.***> 
Verzonden: vrijdag 19 januari 2024 15:42
Aan: Bodmer/TFT_eSPI ***@***.***>
CC: arthurrep ***@***.***>; Author ***@***.***>
Onderwerp: Re: [Bodmer/TFT_eSPI] User_setup for ESP32_2432S022 2.2 inch TFT module (Discussion #3100)

 

In a word... nope.

I spend the best part of a day messing around with this thing and the only way I could get it to display anything was by burning the default bin file from the archive.

Having said that... I did get the backlight to turn on and off!

So... from the supplied circuit diagram there are the following pins -
CS = IO17
RS = IO16
WR = IO4
RD = IO2
RST = TFT_RST

From here I made three assumptions -

RST should be set to -1 because TFT_RST is connected to RST (EN pin)
CS = CS so connect to IO17
As we only have 3 other pins I set MISO to -1 as we don't need data from the display

from there I tried every combination for MOSI,SCK and DC. -

MOSI 16 16 2 2 4 4
SCK 2 4 4 16 16 2
DC 4 2 16 4 2 16

.......... and got nothing.....

 <https://github.com/Bodmer/TFT_eSPI/assets/79755245/d0bda10d-e6f5-499e-a027-b62b4d6ce19b> ESP32-2432022-LCM-V1.0.png (view on web)
 <https://github.com/Bodmer/TFT_eSPI/assets/79755245/73664b00-57b8-4f59-a4b2-14b6250ab675> ESP32-2432022-MCU-V1.0.png (view on web)

Here is my simple test sketch
`//#include <Arduino.h>
#include "TFT_eSPI.h"
//#include <esp_wifi.h>
//#include <esp_event_loop.h>
//#include <nvs_flash.h>

TFT_eSPI tft = TFT_eSPI(240, 380); // Invoke custom library

void setup(){

tft.init();
tft.begin();
tft.setRotation(1);

pinMode(0, OUTPUT);
digitalWrite(0, LOW);
delay(1000);
digitalWrite(0, HIGH);
}

void loop(){
tft.fillScreen(TFT_BLACK);
delay(1000);
tft.fillScreen(TFT_RED);
delay(1000);
tft.fillScreen(TFT_GREEN);
delay(1000);
tft.fillScreen(TFT_BLUE);
delay(1000);
}`

I give up :-)

**Comment by dart1280** [comment]
I haven't really looked at an alternative, I only bought this one because of the price! I'm sure the answer will be simple, it just needs someone with a better understanding of serial protocols to look at it. Having said that I do have a couple of Lilygo TTGO-T-Displays with a 135x240 display that work well....but pricey.
I'm beginning to think the circuit diagrams given are the diagrams for the touch version and my non-touch version might be a revised pcb.

**Reply by dart1280** [reply]
Like I said, it is obvious this needs someone with a better understanding on serial protocols than I possess...however, I have tried again with this setup -

`#define USER_SETUP_ID 999

#define TFT_PARALLEL_8_BIT


#define ST7789_DRIVER
//#define IL19341_DRIVER


// ESP32  pins used for the parallel interface TFT
#define TFT_CS    17
#define TFT_DC    16 // Data Command control pin - must use a GPIO in the range 0-31
#define TFT_RST  -1

#define TFT_WR    4  // Write strobe control pin - must use a GPIO in the range 0-31
#define TFT_RD    2

/*
#define TFT_DB8   15  // Must use GPIO in the range 0-31 for the data bus
#define TFT_DB9   13  // so a single register write sets/clears all bits
#define TFT_DB10  12
#define TFT_DB11  14
#define TFT_DB12  27
#define TFT_DB13  25
#define TFT_DB14  33
#define TFT_DB15  32
*/

  					//Also tried 
#define TFT_D0   15  
#define TFT_D1   13  
#define TFT_D2   12 
#define TFT_D3   14 
#define TFT_D4   27
#define TFT_D5   25
#define TFT_D6   33
#define TFT_D7   32

 /*					//Also tried 
#define TFT_D8   15  
#define TFT_D9   13  
#define TFT_D10  12 
#define TFT_D11  14 
#define TFT_D12  27
#define TFT_D13  25
#define TFT_D14  33
#define TFT_D15  32
*/
/* 					//Also tried 
#define TFT_B0   15  
#define TFT_B1   13  
#define TFT_B2   12 
#define TFT_B3   14 
#define TFT_B4   27
#define TFT_B5   25
#define TFT_B6   33
#define TFT_B7   32
*/


#define LOAD_GLCD
#define LOAD_FONT2
#define LOAD_FONT4
#define LOAD_FONT6
#define LOAD_FONT7
#define LOAD_FONT8
#define LOAD_GFXFF

#define SMOOTH_FONT
`
Here is the circuit diagram of the display - 
![display](https://github.com/Bodmer/TFT_eSPI/assets/79755245/e6981b89-97a0-44c6-a0b6-98fc75e712ba)

I still can't get a display... what am I doing wrong?

**Comment by arthurrep** [comment]
It’s a tall order…

 

I wonder if they show the right display in the documentation. The 2.4” and 2.8” displays have no parallel option… 



 

 

I tried the pin layout below:

#define TFT_SDI 13   // and TFT_MOSI

#define TFT_SCLK 14

//#define TFT_DC -1

#define TFT_RS 2

 

#define TFT_CS 15 

#define TFT_RST -1

define TFT_SDO 12  // and TFT_MISO

 

no result…

 

 

Van: dart1280 ***@***.***> 
Verzonden: maandag 22 januari 2024 18:58
Aan: Bodmer/TFT_eSPI ***@***.***>
CC: arthurrep ***@***.***>; Author ***@***.***>
Onderwerp: Re: [Bodmer/TFT_eSPI] User_setup for ESP32_2432S022 2.2 inch TFT module (Discussion #3100)

 

Like I said, it is obvious this needs someone with a better understanding on serial protocols than I possess...however, I have tried again with this setup -

`#define USER_SETUP_ID 999

#define TFT_PARALLEL_8_BIT

#define ST7789_DRIVER
//#define IL19341_DRIVER

// ESP32 pins used for the parallel interface TFT
#define TFT_CS 17
#define TFT_DC 16 // Data Command control pin - must use a GPIO in the range 0-31
#define TFT_RST -1

#define TFT_WR 4 // Write strobe control pin - must use a GPIO in the range 0-31
#define TFT_RD 2

/*
#define TFT_DB8 15 // Must use GPIO in the range 0-31 for the data bus
#define TFT_DB9 13 // so a single register write sets/clears all bits
#define TFT_DB10 12
#define TFT_DB11 14
#define TFT_DB12 27
#define TFT_DB13 25
#define TFT_DB14 33
#define TFT_DB15 32
*/

                               //Also tried 

#define TFT_D0 15
#define TFT_D1 13
#define TFT_D2 12
#define TFT_D3 14
#define TFT_D4 27
#define TFT_D5 25
#define TFT_D6 33
#define TFT_D7 32

/* //Also tried
#define TFT_D8 15
#define TFT_D9 13
#define TFT_D10 12
#define TFT_D11 14
#define TFT_D12 27
#define TFT_D13 25
#define TFT_D14 33
#define TFT_D15 32
/
/ //Also tried
#define TFT_B0 15
#define TFT_B1 13
#define TFT_B2 12
#define TFT_B3 14
#define TFT_B4 27
#define TFT_B5 25
#define TFT_B6 33
#define TFT_B7 32
*/

#define LOAD_GLCD
#define LOAD_FONT2
#define LOAD_FONT4
#define LOAD_FONT6
#define LOAD_FONT7
#define LOAD_FONT8
#define LOAD_GFXFF

#define SMOOTH_FONT
`
Here is the circuit diagram of the display -
display.jpg (view on web) <https://github.com/Bodmer/TFT_eSPI/assets/79755245/e6981b89-97a0-44c6-a0b6-98fc75e712ba> 

I still can't get a display... what am I doing wrong?

**Reply by dart1280** [reply]
Thanks for the reply.....

Well..... the circuit diagram does have the display connections labelled up as CS, RS, WR and RD so the only anomaly I can see is the one labelled RS which I take to be the same as TFT_DC seeing as it is the only pin not directly attributed to the correct parallel TFT display pins. I also think with DB7 grounded the wiring configures the chip to 8080-1 Series MCU Parallel Interface which is in keeping with the TFT_eSPI library settings.
![TFT_eSPI](https://github.com/Bodmer/TFT_eSPI/assets/79755245/57afa7fe-8ee5-494d-bddf-b8c0a319134b)

I have tidied up my config to make it easier to read.-

#define USER_SETUP_ID 999
#define TFT_PARALLEL_8_BIT  // TFT_eSPI does not support 16 bit parallel
#define ST7789_DRIVER

#define TFT_WIDTH 240
#define TFT_HEIGHT 320

                                              // ESP32  pins used for the parallel interface TFT

#define TFT_CS    17  
#define TFT_DC    16             // Data Command control pin - must use a GPIO in the range 0-31
#define TFT_RST  -1              // TFT_RST is connected to the esp32 enable pin along with the MCU_RST

#define TFT_WR    4             // Write strobe control pin - must use a GPIO in the range 0-31
#define TFT_RD    2

#define TFT_D0   15              // Must use GPIO in the range 0-31 for the data bus
#define TFT_D1   13             // so a single register write sets/clears all bits
#define TFT_D2   12 
#define TFT_D3   14 
#define TFT_D4   27
#define TFT_D5   25
#define TFT_D6   33
#define TFT_D7   32


#define LOAD_GLCD
#define LOAD_FONT2
#define LOAD_FONT4
#define LOAD_FONT6
#define LOAD_FONT7
#define LOAD_FONT8
#define LOAD_GFXFF

#define SMOOTH_FONT
 
![display](https://github.com/Bodmer/TFT_eSPI/assets/79755245/27f2012f-10ee-4454-9f07-629fbf4ac189)

Having broken out the scope I see data on the first 6 pins (D0 to D5) but not on D6 and D7........  more investigations needed ;-)

**Reply by dart1280** [reply]
![All_Traces](https://github.com/Bodmer/TFT_eSPI/assets/79755245/ecb742a1-a2d4-4427-b33a-7b335181c528)

That all looks good I think, the ESP is working, it is just the display that is not. I have checked it all works still with the test .bin supplied with the device and it is all good.

...Hang on... I have just noticed on the cct diagram IM0 is connected to 3v3, meaning the display is configured to 16 bit parallel but the esp32 can't do that......  that makes no sense to me......

**Comment by arthurrep** [comment]
And, does it work now?

 

Van: dart1280 ***@***.***> 
Verzonden: dinsdag 23 januari 2024 12:10
Aan: Bodmer/TFT_eSPI ***@***.***>
CC: arthurrep ***@***.***>; Author ***@***.***>
Onderwerp: Re: [Bodmer/TFT_eSPI] User_setup for ESP32_2432S022 2.2 inch TFT module (Discussion #3100)

 

Thanks for the reply.....

Well..... the circuit diagram does have the display connections labelled up as CS, RS, WR and RD so the only anomaly I can see is the one labelled RS which I take to be the same as TFT_DC seeing as it is the only pin not directly attributed to the correct parallel TFT display pins. I also think with DB7 grounded the wiring configures the chip to 8080-1 Series MCU Parallel Interface which is in keeping with the TFT_eSPI library settings.
TFT_eSPI.jpg (view on web) <https://github.com/Bodmer/TFT_eSPI/assets/79755245/57afa7fe-8ee5-494d-bddf-b8c0a319134b> 

I have tidied up my config to make it easier to read.-

#define USER_SETUP_ID 999
#define TFT_PARALLEL_8_BIT // TFT_eSPI does not support 16 bit parallel
#define ST7789_DRIVER

#define TFT_WIDTH 240
#define TFT_HEIGHT 320

                                          // ESP32  pins used for the parallel interface TFT

#define TFT_CS 17
#define TFT_DC 16 // Data Command control pin - must use a GPIO in the range 0-31
#define TFT_RST -1 // TFT_RST is connected to the esp32 enable pin along with the MCU_RST

#define TFT_WR 4 // Write strobe control pin - must use a GPIO in the range 0-31
#define TFT_RD 2

#define TFT_D0 15 // Must use GPIO in the range 0-31 for the data bus
#define TFT_D1 13 // so a single register write sets/clears all bits
#define TFT_D2 12
#define TFT_D3 14
#define TFT_D4 27
#define TFT_D5 25
#define TFT_D6 33
#define TFT_D7 32

#define LOAD_GLCD
#define LOAD_FONT2
#define LOAD_FONT4
#define LOAD_FONT6
#define LOAD_FONT7
#define LOAD_FONT8
#define LOAD_GFXFF

#define SMOOTH_FONT

display.jpg (view on web) <https://github.com/Bodmer/TFT_eSPI/assets/79755245/27f2012f-10ee-4454-9f07-629fbf4ac189>

**Reply by dart1280** [reply]
No.. however I have made progress, I have the display working using the LGFX library.  (https://github.com/lovyan03/LovyanGFX)


```
#include <Arduino.h>
#define TFT_WIDTH 240
#define TFT_HEIGHT 320
#define LGFX_USE_V1      // set to use new version of library
#include <LovyanGFX.hpp> // main library

class LGFX : public lgfx::LGFX_Device
{

    lgfx::Panel_ST7789 _panel_instance; // ST7789UI
    lgfx::Bus_Parallel8 _bus_instance;  // MCU8080 8B

public:
    LGFX(void)
    {
        {
            auto cfg = _bus_instance.config();
            cfg.freq_write = 25000000;
            cfg.pin_wr = 4;
            cfg.pin_rd = 2;
            cfg.pin_rs = 16;

            cfg.pin_d0 = 15;
            cfg.pin_d1 = 13;
            cfg.pin_d2 = 12;
            cfg.pin_d3 = 14;
            cfg.pin_d4 = 27;
            cfg.pin_d5 = 25;
            cfg.pin_d6 = 33;
            cfg.pin_d7 = 32;

            _bus_instance.config(cfg);
            _panel_instance.setBus(&_bus_instance);
        }

        {
            auto cfg = _panel_instance.config();

            cfg.pin_cs = 17;
            cfg.pin_rst = -1;
            cfg.pin_busy = -1;

            cfg.panel_width = 240;
            cfg.panel_height = 320;
            cfg.offset_x = 0;
            cfg.offset_y = 0;
            cfg.offset_rotation = 0;
            cfg.readable = false;
            cfg.invert = false;
            cfg.rgb_order = false;
            cfg.dlen_16bit = false;
            cfg.bus_shared = true;

            _panel_instance.config(cfg);
        }

        setPanel(&_panel_instance);
    }
};

static LGFX tft; // declare display variable


void setup()
{
    Serial.begin(115200);

    pinMode(0, OUTPUT);

    digitalWrite(0, LOW);
    delay(1000);
    digitalWrite(0, HIGH);

    tft.init();
    tft.fillScreen(TFT_YELLOW);
    delay(500);
    tft.fillScreen(TFT_RED);
    delay(500);
    tft.fillScreen(TFT_GREEN);
    delay(500);
    tft.fillScreen(TFT_BLUE);
    delay(500);
  
}

void loop()
{
   tft.fillScreen(TFT_BLACK);
    delay(1000);
    tft.fillScreen(TFT_RED);
    delay(1000);
    tft.fillScreen(TFT_GREEN);

    delay(1000);
    tft.fillScreen(TFT_BLUE);
    delay(1000);
    Serial.println("end of loop");
}
```
So transposing the values for CS,RST, RS, WR,RD and BL (which are the same names in the TFT_eSPI library) and setting the remaining TFT_CD to -1 (DC doesn't exist in the circuit diagram either) ...and......... nope, it still refuses to work
```
#define TFT_PARALLEL_8_BIT  // TFT_eSPI does not support 16 bit parallel
#define ST7789_DRIVER

#define TFT_WIDTH 240
#define TFT_HEIGHT 320

// ESP32  pins used for the parallel interface TFT


				//This pin is not defined in the working LGFX library and is not shown on the schematic

#define TFT_DC   -1 // Data Command control pin - must use a GPIO in the range 0-31

 
				//these pins are as defined in the working LGFX library
				
#define TFT_CS   17 				
#define TFT_RST  -1  // TFT_RST is connected to the esp32 enable pin along with the MCU_RST
#define TFT_RS   16
#define TFT_WR    4  // Write strobe control pin - must use a GPIO in the range 0-31
#define TFT_RD    2
#define TFT_BL    0




#define TFT_D0   15   // Must use GPIO in the range 0-31 for the data bus
#define TFT_D1   13   // so a single register write sets/clears all bits
#define TFT_D2   12 
#define TFT_D3   14 
#define TFT_D4   27
#define TFT_D5   25
#define TFT_D6   33
#define TFT_D7   32


#define LOAD_GLCD
#define LOAD_FONT2
#define LOAD_FONT4
#define LOAD_FONT6
#define LOAD_FONT7
#define LOAD_FONT8
#define LOAD_GFXFF

#define SMOOTH_FONT

#define SPI_FREQUENCY  25000000
#define SPI_READ_FREQUENCY  20000000
//#define USE_HSPI_PORT
```

I see the same company offer a 2.8" display that doesn't use parallel addressing and they supply a setup file for it so I'm going to give up on this one and await the arrival of the 2.8" display ;-)

**Comment by arthurrep** [comment]
Good morrning,

 

I tried the GFX-code on my computer but I get the following error message :

 

d:\Mijn bestanden\Documents\Arduino\libraries\LovyanGFX-master\src\lgfx\v1\platforms\esp32\Bus_SPI.cpp: In member function 'virtual bool lgfx::v1::Bus_SPI::init()':

d:\Mijn bestanden\Documents\Arduino\libraries\LovyanGFX-master\src\lgfx\v1\platforms\esp32\Bus_SPI.cpp:164:20: error: 'DPORT_SPI_DMA_CHAN_SEL_REG' was not declared in this scope

     dma_ch = (*reg(DPORT_SPI_DMA_CHAN_SEL_REG) >> (_cfg.spi_host * 2)) & 3;

                    ^~~~~~~~~~~~~~~~~~~~~~~~~~

d:\Mijn bestanden\Documents\Arduino\libraries\LovyanGFX-master\src\lgfx\v1\platforms\esp32\Bus_SPI.cpp:164:20: note: suggested alternative: 'SOC_SPI_DMA_CHAN_NUM'

     dma_ch = (*reg(DPORT_SPI_DMA_CHAN_SEL_REG) >> (_cfg.spi_host * 2)) & 3;

                    ^~~~~~~~~~~~~~~~~~~~~~~~~~

                    SOC_SPI_DMA_CHAN_NUM

Multiple libraries were found for "SPI.h"

  Used: C:\Users\Gebruiker\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.11\libraries\SPI

  Not used: D:\Mijn bestanden\Documents\Arduino\libraries\SPI

Multiple libraries were found for "Wire.h"

  Used: C:\Users\Gebruiker\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.11\libraries\Wire

  Not used: D:\Mijn bestanden\Documents\Arduino\libraries\Wire

exit status 1

 

this also occurs also when I try the factory samples…

 

 

Compilation error: exit status 1

 

Van: dart1280 ***@***.***> 
Verzonden: dinsdag 23 januari 2024 22:38
Aan: Bodmer/TFT_eSPI ***@***.***>
CC: arthurrep ***@***.***>; Author ***@***.***>
Onderwerp: Re: [Bodmer/TFT_eSPI] User_setup for ESP32_2432S022 2.2 inch TFT module (Discussion #3100)

 

No.. however I have made progress, I have the display working using the LGFX library. (https://github.com/lovyan03/LovyanGFX)

#include <Arduino.h>
#define TFT_WIDTH 240
#define TFT_HEIGHT 320
#define LGFX_USE_V1      // set to use new version of library
#include <LovyanGFX.hpp> // main library
 
class LGFX : public lgfx::LGFX_Device
{
 
    lgfx::Panel_ST7789 _panel_instance; // ST7789UI
    lgfx::Bus_Parallel8 _bus_instance;  // MCU8080 8B
 
public:
    LGFX(void)
    {
        {
            auto cfg = _bus_instance.config();
            cfg.freq_write = 25000000;
            cfg.pin_wr = 4;
            cfg.pin_rd = 2;
            cfg.pin_rs = 16;
 
            cfg.pin_d0 = 15;
            cfg.pin_d1 = 13;
            cfg.pin_d2 = 12;
            cfg.pin_d3 = 14;
            cfg.pin_d4 = 27;
            cfg.pin_d5 = 25;
            cfg.pin_d6 = 33;
            cfg.pin_d7 = 32;
 
            _bus_instance.config(cfg);
            _panel_instance.setBus(&_bus_instance);
        }
 
        {
            auto cfg = _panel_instance.config();
 
            cfg.pin_cs = 17;
            cfg.pin_rst = -1;
            cfg.pin_busy = -1;
 
            cfg.panel_width = 240;
            cfg.panel_height = 320;
            cfg.offset_x = 0;
            cfg.offset_y = 0;
            cfg.offset_rotation = 0;
            cfg.readable = false;
            cfg.invert = false;
            cfg.rgb_order = false;
            cfg.dlen_16bit = false;
            cfg.bus_shared = true;
 
            _panel_instance.config(cfg);
        }
 
        setPanel(&_panel_instance);
    }
};
 
static LGFX tft; // declare display variable
 
 
void setup()
{
    Serial.begin(115200);
 
    pinMode(0, OUTPUT);
 
    digitalWrite(0, LOW);
    delay(1000);
    digitalWrite(0, HIGH);
 
    tft.init();
    tft.fillScreen(TFT_YELLOW);
    delay(500);
    tft.fillScreen(TFT_RED);
    delay(500);
    tft.fillScreen(TFT_GREEN);
    delay(500);
    tft.fillScreen(TFT_BLUE);
    delay(500);
  
}
 
void loop()
{
   tft.fillScreen(TFT_BLACK);
    delay(1000);
    tft.fillScreen(TFT_RED);
    delay(1000);
    tft.fillScreen(TFT_GREEN);
 
    delay(1000);
    tft.fillScreen(TFT_BLUE);
    delay(1000);
    Serial.println("end of loop");
}

So transposing the values for CS,RST, RS, WR,RD and BL (which are the same names in the TFT_eSPI library) and setting the remaining TFT_CD to -1 (DC doesn't exist in the circuit diagram either) ...and......... nope, it still refuses to work

#define TFT_PARALLEL_8_BIT  // TFT_eSPI does not support 16 bit parallel
#define ST7789_DRIVER
 
#define TFT_WIDTH 240
#define TFT_HEIGHT 320
 
// ESP32  pins used for the parallel interface TFT
 
 
                               //This pin is not defined in the working LGFX library and is not shown on the schematic
 
#define TFT_DC   -1 // Data Command control pin - must use a GPIO in the range 0-31
 
 
                               //these pins are as defined in the working LGFX library
                               
#define TFT_CS   17                           
#define TFT_RST  -1  // TFT_RST is connected to the esp32 enable pin along with the MCU_RST
#define TFT_RS   16
#define TFT_WR    4  // Write strobe control pin - must use a GPIO in the range 0-31
#define TFT_RD    2
#define TFT_BL    0
 
 
 
 
#define TFT_D0   15   // Must use GPIO in the range 0-31 for the data bus
#define TFT_D1   13   // so a single register write sets/clears all bits
#define TFT_D2   12 
#define TFT_D3   14 
#define TFT_D4   27
#define TFT_D5   25
#define TFT_D6   33
#define TFT_D7   32
 
 
#define LOAD_GLCD
#define LOAD_FONT2
#define LOAD_FONT4
#define LOAD_FONT6
#define LOAD_FONT7
#define LOAD_FONT8
#define LOAD_GFXFF
 
#define SMOOTH_FONT
 
#define SPI_FREQUENCY  25000000
#define SPI_READ_FREQUENCY  20000000
//#define USE_HSPI_PORT

**Comment by dart1280** [comment]
That looks like a library revision issue to me, The sketch compiles ok for me, I have the following revisions - 
Arduino IDE 2.2.1
Using library LovyanGFX at version 1.1.12 in folder: E:\data\Documents\Arduino\libraries\LovyanGFX 
Using library SPI at version 2.0.0 in folder: C:\Users\****\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.14\libraries\SPI 
Using library Wire at version 2.0.0 in folder: C:\Users\****\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.14\libraries\Wire 

Perhaps check what revisions you have? Your ESP32 library is older than mine....


My project :-)
![IMG_20240125_104356](https://github.com/Bodmer/TFT_eSPI/assets/79755245/7efd0ef7-fde2-40d0-87f5-79da7388677f)

**Comment by arthurrep** [comment]
Updating the LovyanGFX library was the solution. Thanks!

 

 

Van: dart1280 ***@***.***> 
Verzonden: donderdag 25 januari 2024 10:40
Aan: Bodmer/TFT_eSPI ***@***.***>
CC: arthurrep ***@***.***>; Author ***@***.***>
Onderwerp: Re: [Bodmer/TFT_eSPI] User_setup for ESP32_2432S022 2.2 inch TFT module (Discussion #3100)

 

That looks like a library revision issue to me, The sketch compiles ok for me, I have the following revisions -
Arduino IDE 2.2.1
Using library LovyanGFX at version 1.1.12 in folder: E:\data\Documents\Arduino\libraries\LovyanGFX
Using library SPI at version 2.0.0 in folder: C:\Users*\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.14\libraries\SPI
Using library Wire at version 2.0.0 in folder: C:\Users*\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.14\libraries\Wire

Perhaps check what revisions you have?

**Comment by arthurrep** [comment]
Where can I find all the commands for using the GFX library?

 

Van: Arthur Rep ***@***.***> 
Verzonden: donderdag 25 januari 2024 12:36
Aan: 'Bodmer/TFT_eSPI' ***@***.***>
Onderwerp: RE: [Bodmer/TFT_eSPI] User_setup for ESP32_2432S022 2.2 inch TFT module (Discussion #3100)

 

Updating the LovyanGFX library was the solution. Thanks!

 

 

Van: dart1280 ***@***.*** ***@***.***> > 
Verzonden: donderdag 25 januari 2024 10:40
Aan: Bodmer/TFT_eSPI ***@***.*** ***@***.***> >
CC: arthurrep ***@***.*** ***@***.***> >; Author ***@***.***>
Onderwerp: Re: [Bodmer/TFT_eSPI] User_setup for ESP32_2432S022 2.2 inch TFT module (Discussion #3100)

 

That looks like a library revision issue to me, The sketch compiles ok for me, I have the following revisions -
Arduino IDE 2.2.1
Using library LovyanGFX at version 1.1.12 in folder: E:\data\Documents\Arduino\libraries\LovyanGFX
Using library SPI at version 2.0.0 in folder: C:\Users*\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.14\libraries\SPI
Using library Wire at version 2.0.0 in folder: C:\Users*\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.14\libraries\Wire

Perhaps check what revisions you have?

**Comment by dart1280** [comment]
I don't know, normally there is a keywords file in the download archive but this library doesn't seem to have one. I got the ones I needed from the example sketches  If you hover over a keyword a pop up box gives you the parameters needed. It appears to be similar to the AdafruitGFX library so try some of those keywords. I guess the writers think you already know them!

**Comment by arthurrep** [comment]
Hi,

 

The AdafruitGFX library is very useful. 

Thanks!

 

 

 

Van: dart1280 ***@***.***> 
Verzonden: vrijdag 26 januari 2024 09:49
Aan: Bodmer/TFT_eSPI ***@***.***>
CC: arthurrep ***@***.***>; Author ***@***.***>
Onderwerp: Re: [Bodmer/TFT_eSPI] User_setup for ESP32_2432S022 2.2 inch TFT module (Discussion #3100)

 

I don't know, normally there is a keywords file in the download archive but this library doesn't seem to have one. I got the ones I needed from the example sketches If you hover over a keyword a pop up box gives you the parameters needed. It appears to be similar to the AdafruitGFX library so try some of those keywords. I guess the writers think you already know them!

**Comment by zealcrew** [comment]
Hello everybody! Maybe this experience will help someone.
Initializing the display with the LovyanGFX library from the ESP32-2432s022 examples and applying the demo from the ESP32-2432S028R in a hurry. https://github.com/zealcrew/esp32-2432s022

**Comment by dart1280** [comment]
I ended up using the LGFX lib for the application I had. I didn't need the
touch so never looked into it. As things have progressed the 2.8 inch
version has much better 'legs' than the 2.2 inch and that works fine with
eSPI so most people would use that display I guess.
